compiler/ballerina-lang/src/main/java/org/wso2/ballerinalang/compiler/semantics/analyzer/TypeChecker.java

```diff
@@ -1764,18 +1761,26 @@ public void visit(BLangLetExpression letExpression) {
     private void checkInLangLib(BLangInvocation iExpr, BType varRefType) {
         boolean langLibMethodExists = checkLangLibMethodInvocationExpr(iExpr, varRefType);
         if (!langLibMethodExists) {
-            dlog.error(iExpr.name.pos, DiagnosticCode.UNDEFINED_FUNCTION, iExpr.name.value);
+            dlog.error(iExpr.name.pos, DiagnosticCode.UNDEFINED_FUNCTION_IN_TYPE, iExpr.name.value, iExpr.expr.type);
             resultType = symTable.semanticError;
         }
     }
 
     private boolean checkFieldFunctionPointer(BLangInvocation iExpr) {
-        BType type = checkExpr(iExpr.expr, this.env);
+        BLangExpression invocationExpr = iExpr.expr;
+        BLangIdentifier invocationIdentifier = iExpr.name;
+
+        BType type = checkExpr(invocationExpr, this.env);
```

Comment:
Instead of using `this.env`, shall we pass the relevant `env` to the function?

Reply:
Done